pkg/asset/store.go

```diff
@@ -25,6 +33,41 @@ func (s *StoreImpl) Fetch(asset Asset) error {
 	return s.fetch(asset, "")
 }
 
+// Load retrieves the state from the state file present in the given directory
+// and returns the assets map
+func (s *StoreImpl) Load(dir string) error {
+	path := filepath.Join(dir, stateFileName)
+	assets := make(map[string][]*File)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(data, &assets)
+}
+
+// Save dumps the entire state map into a file
+func (s *StoreImpl) Save(dir string) error {
+	assetMap := make(map[string][]*File)
```

Comment:
This seems contradictory to what @staebler recommended. the state file stores the actual type not just files.

Reply:
It does store the actual type only. In the form of a string.

Reply:
Can you start documentation why we chose `The non-writable assets do not go into the state file`.